### ISSUE TITLE
git-crypt unlock failing 

### ISSUE THREAD

**Issue by russellsimpkins** [AGWA/git-crypt#52]
I have a git repo on bitbucket. I was able to configure git-crypt add secure files lock and unlock. I recently tried cloning the repo and the running git-crypt unlock and it is failing.

```
> git status
On branch master
Your branch is up-to-date with 'origin/master'.
nothing to commit, working directory clean

> git-crypt unlock

You need a passphrase to unlock the secret key for
user: "Russell Simpkins (Russ) <russellsimpkins@donotreply>"
4096-bit RSA key, ID 065C5B7A, created 2015-04-20 (main key ID 3B3F5A08)

git-crypt: error: encrypted file has been tampered with!
error: external filter "git-crypt" smudge failed 1
error: external filter "git-crypt" smudge failed
fatal: files/secure/appstack/app.properties: smudge filter git-crypt failed
Error: 'git checkout' failed
git-crypt has been set up but existing encrypted files have not been decrypted
```
Here is my current config
```
> cat .gitattributes 
files/secure/**/* filter=git-crypt diff=git-crypt

> git-crypt status |grep files/secure
    encrypted: files/secure/appstack/app.properties
    encrypted: files/secure/jenkins/id_jenkins
```

I thought maybe having the wildcards was causing an issue. I tried pointing at the files e.g.

```
> cat .gitattributes 
files/secure/appstack/app.properties filter=git-crypt diff=git-crypt
files/secure/jenkins/id_jenkins filter=git-crypt diff=git-crypt
```
But that did not make a difference. 

I'm working on Yosemite 10.10.3. Git-crypt installed using brew  to install version 0.4.2

Any thoughts on this? Is it possibly related to bitbucket?

**Comment by AGWA** [comment]
1. Does `git log files/secure/appstack/app.properties` indicate any commit where the file may have been modified inadvertently?

2. In the original repository, where app.properties is checked out unencrypted, run `touch files/secure/appstack/app.properties`, and then run `git status` to see if it's reported as modified.  If it is, commit it and then try pulling it down to the other repository and unlocking.

3. Does `git check-attr -a files/secure/appstack/app.properties` show any attributes besides git-crypt, in either the original repository or the clone?

**Comment by russellsimpkins** [comment]
Touching the files, committing them and then ``` git-crypt unlock ``` allowed me to decrypt the data.

**Comment by AGWA** [comment]
After that file was initially committed to the repository in encrypted form, the encrypted version was modified somehow, resulting in the "encrypted file has been tampered with" error.  (This is why I suggested checking `git log` for inadvertent modifications.)  Touching the file and re-committing it forces git-crypt to re-encrypt it.

**Comment by russellsimpkins** [comment]
Very good to know a touch && commit lets me unlock the data. Thank you.

**Comment by russellsimpkins** [comment]
Actually, touching && commit does not fix the issue. I can now run git-crypt unlock, but the data is encrypted. So, it looks like git-crypt re-encrypted the encrypted file.

**Comment by AGWA** [comment]
Is the file encrypted or decrypted in the original repository?

**Comment by russellsimpkins** [comment]
If you clone the repo, the data is encrypted.

I've gone to the trouble to checkout the revision when i initially added the data files and the unlock them and it's no use. I still get the error.

**Comment by russellsimpkins** [comment]
once i've encrypted the data, do you have to ``` git-crypt lock ``` before you push? 
if i put the repo back to the commit when a file was added, shouldn't that fix any possible future commit flubs? Are there other hidden files that need to be added to restore from such a problem?

**Comment by AGWA** [comment]
No, you do not need to run `git-crypt lock` before pushing.

I recommend you do the following:
1. Ensure the repo is unlocked by running `git-crypt unlock`
2. Ensure all the files are in unencrypted form.  Replace their contents if necessary.  Then touch and commit them.
3. Ensure you can clone the repository and unlock the clone.

**Comment by russellsimpkins** [comment]
I'm baffled. I restored the secure files and did a bunch of testing and can not re-produce my issue. So, there must have been some disconnect on my end. Thank you for taking the time to help and answer my questions.

**Comment by skorokithakis** [comment]
I'm seeing the same. Touching and re-committing doesn't fix it. Even if I check the repo out in a different dir on the same machine, `git-crypt unlock` chokes, even though the original repo is happy as a clam. I don't know what could be wrong at this point.

**Comment by AGWA** [comment]
@skorokithakis Could you open a new issue for this, describing the exact sequence of operations you're doing?  Thanks!

**Comment by skorokithakis** [comment]
I could, but I'm afraid I really don't have any info apart from `git checkout <url>; git unlock; <disaster>`. I don't even know *why* it does it.

Regardless, I'll open a new ticket, thanks.

**Comment by nakanaa** [comment]
Just in case someone ends up here from Google like I did, I'll add in my experience:
I had this exact same error (using newest version 0.6.0, trying to unlock a repo with a symmetric key) and used a couple hours to fight with this. In the end I learned the key I had actually wasn't the right key for the repo like I had thought it was. So **using the wrong key to unlock gave me this error**.

So make sure you have the right key. The error message really isn't clear that this is a possible reason for the failure.
